botocore/awsrequest.py

```diff
@@ -450,6 +471,92 @@ def prepare_body(self, data, files, json=None):
                 if 'Transfer-Encoding' in self.headers:
                     self.headers.pop('Transfer-Encoding')
 
+        if self.body and 'Content-Length' not in self.headers:
+            self.headers['Transfer-Encoding'] = 'chunked'
+
+
+class AWSResponse(object):
+
+    def __init__(self, url=None, status_code=None, headers=None, raw=None):
+        self.url = url
+        self.status_code = status_code
+        self.headers = HeadersDict(headers or {})
+        self.raw = raw
+
+        self._content = None
+
+    @property
+    def content(self):
+        """Content of the response, in bytes."""
+
+        if self._content is None:
+            # Read the contents.
+            try:
+                data_generator = self.raw.stream()
+            except AttributeError:
```

Comment:
When would `raw` not have a stream method on it?

Reply:
When it's not a raw response  from urllib3. 

Currently, this will never happen in practice but we relied on this functionality from requests in some tests where we mock the raw response with a file-like object.

Reply:
Yeah I'd prefer to simplify and remove this fallback if our tests are the only thing relying on it.

Reply:
I removed it and just updated the tests that were dependent on it.